jbpm-flow/src/main/java/org/jbpm/ruleflow/core/validation/RuleFlowProcessValidator.java

```diff
@@ -355,12 +355,16 @@ else if (node instanceof SubProcessNode) {
                 validateNodes(forEachNode.getNodes(), errors, process);
             } else if (node instanceof DynamicNode) {
                 final DynamicNode dynamicNode = (DynamicNode) node;
-                if (dynamicNode.getDefaultIncomingConnections().size() == 0) {
-                    addErrorMessage(process, node, errors, "Dynamic has no incoming connection");
-                }
-                if (dynamicNode.getDefaultOutgoingConnections().size() == 0) {
-                    addErrorMessage(process, node, errors, "Dynamic has no outgoing connection");
+
+                if(!process.isDynamic()) {
+                    if (dynamicNode.getDefaultIncomingConnections().size() == 0) {
```

Comment:
I noticed that the previous checks use acceptsNoIncomingConnections(dynamicNode) so we might use that one.